openssl-dynamic/src/main/java/org/apache/tomcat/jni/socket/AprSocketContext.java

```diff
@@ -797,10 +797,16 @@ void prepare() throws IOException {
         }
 
         void unblock() {
-            try (java.net.Socket sock = new java.net.Socket()) {
-                // Easiest ( maybe safest ) way to interrupt accept
-                // we could have it in non-blocking mode, etc
-                sock.connect(new InetSocketAddress("127.0.0.1", port));
+            java.net.Socket sock = new java.net.Socket();
+
+            try {
+                try {
+                    // Easiest ( maybe safest ) way to interrupt accept
+                    // we could have it in non-blocking mode, etc
+                    sock.connect(new InetSocketAddress("127.0.0.1", port));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this IP "127.0.0.1" address configurable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1313)